### ISSUE TITLE
dotnet configuration unsucessful

### ISSUE THREAD

**Issue by zenayee** [dotnet/website-feedback#298]
### URL

https://dotnet.microsoft.com/en-us/learn/dotnet/hello-world-tutorial/install

### Operating system

windows

### Description

 The configuration unit failed while attempting to apply the desired state.
Executing 'C:\Users\ZYEE1\AppData\Local\Programs\Microsoft VS Code\bin\code.cmd --install-extension ms-dotnettools.csdevkit' failed. Command Output: 'Error while installing extensions: getaddrinfo ENOTFOUND marketplace.visualstudio.com\n getaddrinfo ENOTFOUND marketplace.visualstudio.com\n '
<See the log file for additional details>
Some of the configuration was not applied successfully.

### dotnet --info

no output after running dotnet command.

**Comment by katiesavage** [comment]
Looks like there may have been an issue with the Winget file installing C# Dev Kit. @claudiaregio have you seen this before?

**Comment by denelon** [comment]
Can you share the details for WinGet like the output from `winget --info`? Did you try running the configuration a second time? It looks like the DSC Resource for installing the extension encountered some kind of an error.

**Comment by zenayee** [comment]
***@***.***

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by zenayee** [comment]
Hello,

What is the status of this issue? Any other info you need from me?

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

From: Yee, Zena (Z.Y.) ***@***.***>
Sent: Tuesday, December 3, 2024 11:46 AM
To: dotnet/website-feedback ***@***.***>; dotnet/website-feedback ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [dotnet/website-feedback] dotnet configuration unsucessful (Issue #298)

***@***.***

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by claudiaregio** [comment]
@zenayee We have asked for additional information in order to be able to debug. Can you please run `winget --info` and provide the logs to us? We cannot troubleshoot any further without additional information

**Comment by zenayee** [comment]
Please see attached email. Thanks!

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by claudiaregio** [comment]
@zenayee I am not seeing anything attached. 

It would be great if you could follow these steps: 
1. winget configure <your path to 'dotnet_configuration.winget'> --verbose --logs
2. Select all the files that are generated and compress as zip file
3. Put that file in this issue

**Comment by zenayee** [comment]
***@***.***

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by zenayee** [comment]
What is <your path to 'dotnet_configuration.winget'>?

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by claudiaregio** [comment]
@zenayee The path is the location on your device where the configuration file is stored. 

If the file is still in your downloads folder, I would recommend right-clicking on your downloads folder, selecting "Open in terminal", and then running `winget configure dotnet_configuration.winget --verbose --logs`.

**Comment by zenayee** [comment]
Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by denelon** [comment]
@zenayee, the replies via e-mail are showing up a bit garbled at GitHub. Please use https://github.com/dotnet/website-feedback/issues/298 to attach logs and provide details on what's happening.

**Comment by zenayee** [comment]
[WinGet-2024-12-13-11-47-44.710.log](https://github.com/user-attachments/files/18129589/WinGet-2024-12-13-11-47-44.710.log)
[WinGet-2024-12-13-11-48-52.496.log](https://github.com/user-attachments/files/18129590/WinGet-2024-12-13-11-48-52.496.log)

**Comment by denelon** [comment]
@zenayee it looks like the command being executed is the path to a directory rather than a path to a configuration file.

The path including the filename should be passed to `winget configure`

**Comment by zenayee** [comment]
[WinGet-2024-12-13-13-00-09.294.log](https://github.com/user-attachments/files/18129772/WinGet-2024-12-13-13-00-09.294.log)
[WinGetCOM-2024-12-13-13-01-05.991.log](https://github.com/user-attachments/files/18129773/WinGetCOM-2024-12-13-13-01-05.991.log)

**Comment by denelon** [comment]
OK, I see what's happening now. @claudiaregio is going to provide an updated configuration file.

On your system if you remove "allowPrerelease: true" under the line with "description: Install Visual Studio Code" it should work. There was an update we performed, and it's in a prerelease module that has a bug in it.

**Comment by denelon** [comment]
Thank you so much for reporting this!

**Comment by claudiaregio** [comment]
@zenayee Thank you so much for helping us debug! 

Please do the following: 
1. Clear contents in `%LOCALAPPDATA%\Microsoft\WinGet\Configuration\Modules` 
2. Run this file and continue with tutorial :) [dotnet_configuration_new.zip](https://github.com/user-attachments/files/18131703/dotnet_configuration_new.zip)

Please let me if this works for you!

**Comment by zenayee** [comment]
Sorry for the late response. I was out of office during the past few weeks. I tried the following steps and got this error message.

***@***.***

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by zenayee** [comment]
Hello,

What is the status of the following issue?

Thanks,

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

From: Yee, Zena (Z.Y.) ***@***.***>
Sent: Thursday, January 2, 2025 10:23 AM
To: dotnet/website-feedback ***@***.***>; dotnet/website-feedback ***@***.***>
Cc: Mention ***@***.***>; github-actions[bot] ***@***.***>
Subject: RE: [dotnet/website-feedback] dotnet configuration unsucessful (Issue #298)

Sorry for the late response. I was out of office during the past few weeks. I tried the following steps and got this error message.

***@***.***

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by zenayee** [comment]
Hi,

I am still experiencing issues after the new instruction.  Please see the attached email for details.

Thanks,


Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by claudiaregio** [comment]
@zenayee Thanks! I am able to see that. I've asked someone to take a look at it, in the meantime just wanted to confirm that you cleared out the contents of `%LOCALAPPDATA%\Microsoft\WinGet\Configuration\Modules` before running the new config file? 

I'd like to unblock you in the meantime, so please follow the instruction below and continue with the tutorial! 
![Image](https://github.com/user-attachments/assets/b0f7a4b4-0952-4bac-8c8b-aa6219e12d61)


If you can still help us debug so we can make the experience better for other users it would be great if you could follow similar instructions and get us the logs. Since you're running into a new issue we need to see what is going on now. Thanks again for your assistance! 

1. Clear contents of `%LOCALAPPDATA%\Microsoft\WinGet\Configuration\Modules`
2. Run [dotnet_configuration_new.zip](https://github.com/user-attachments/files/18131703/dotnet_configuration_new.zip)
3. If the config file is in your downloads folder...right-click on your downloads folder, select "Open in terminal", and then run winget configure dotnet_configuration.winget --verbose --logs`
4. Paste the logs in this issue via GitHub (not email)

**Comment by zenayee** [comment]
Yes, I cleared all the content in %LOCALAPPDATA%\Microsoft\WinGet\Configuration\Modules last time I ran the configuration. And I did it this time as well.

![Image](https://github.com/user-attachments/assets/c4cf1947-e01b-41d9-94a0-661ec5a70d1f)

**Comment by zenayee** [comment]
One more question. Should the step 3 in your above instruction be dotnet_configuration_**new**.winget instead of dotnet_configuration.winget?

**Comment by zenayee** [comment]
If I use dotnet_configuration_new.winget the output is as follows.

![Image](https://github.com/user-attachments/assets/4e25b457-d353-4a3f-a23f-726422fe5107)

**Comment by zenayee** [comment]
Hi,

Is there any progress on this issue? I really need it be fixed to perform some of my work in the next 2 weeks. Thanks!

Zena Yee
Product Development Engineer
Closed Loop Fuel Features
Vehicle Controls
E-mail: ***@***.******@***.***>
Tel: (313) 585-9655

**Comment by claudiaregio** [comment]
Hi @zenayee , yes good catch apologies about that. You should be running the logs command with the new file (dotnet_configuration_new.winget). Unfortunately we cannot debug with just the error on the screen, we need the actual logs of that run like you did on Dec 13 👇

> [WinGet-2024-12-13-11-47-44.710.log](https://github.com/user-attachments/files/18129589/WinGet-2024-12-13-11-47-44.710.log) [WinGet-2024-12-13-11-48-52.496.log](https://github.com/user-attachments/files/18129590/WinGet-2024-12-13-11-48-52.496.log)

### We really really appreciate you helping us by providing info so we can make sure others don't run into this but I have provided instructions on how to unblock yourself within VS Code so you can get to work! It appears the config file successfully installed the .NET 8 SDK and VS Code, all you need that's left is to install the C# Dev Kit extension and you can do this from VS Code itself! Please follow picture below

> I'd like to unblock you in the meantime, so please follow the instruction below and continue with the tutorial! ![Image](https://github.com/user-attachments/assets/b0f7a4b4-0952-4bac-8c8b-aa6219e12d61)

**Comment by denelon** [comment]
@zenayee are you still experiencing trouble with WinGet Configuration?

We've had a couple of new releases, and VSCode has also had several new releases. I'd love to get to the bottom of this one.
